functions-app/school-results-load-cache/v1.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+const RA = require('ramda-adjunct')
+const schoolService = require('./services/school.service')
+const mapLimit = require('async/mapLimit')
+
+const name = 'school-results-load-cache:v1'
+const asyncLimit = 5 // number of async operations to do at a time
+
+const v1 = {
+  process: async (context) => {
+    const logger = context.log
+    const schools = await schoolService.retrieveAll()
+    if (RA.isArray(schools) && schools.length) {
+      logger(`${name} ${schools.length} schools to process`)
+      schoolService.setLogger(context.log)
+      /** @var { {name:string, id:number, error:undefined|Error success:boolean}[] } result **/
+      const result = await mapLimit(schools, asyncLimit, schoolService.processOne)
```

Comment:
going back to our conversation on `mapLimit` earlier in the week - this will still process the rest of the schools even if one fails?

Reply:
yes - it catches errors and returns an object detailing that processing for that school failed with the error.